### DISCUSSION TITLE
UI Tweaks

### DISCUSSION THREAD

**Discussion by jstars-gmti** [netalertx/NetAlertX#904] [Ideas]
This is low priority, but I think that the UI could benefit from standardizing a bit more with Pi-hole. Pi-hole has an excellent UI that shows just whats important. 

**Dashboard Metrics**: The netalertx dash board displays too much info making it quite wide. This contrasts with Pi-hole which has only four main blocks in four colors.

Pi-Hole:
<img width="1201" alt="Screenshot 2024-12-03 at 09 48 14" src="https://github.com/user-attachments/assets/38ff2fc0-21ef-43ca-b5aa-17ec464486ad">

NetAlertX:
<img width="1200" alt="Screenshot 2024-12-03 at 10 22 41" src="https://github.com/user-attachments/assets/8de9de2f-84bf-47ee-a417-54dc2f3169d4">
The purpose of a dashboard is to show important information at a glance. I submit that the most important info is **Devices**, **Connected**, **Down**, and **New Devices**. My advice would be to remove **Favorites** and **Archived** (these are available from the menubar on the left anyway) from the dashboard block, leaving the four most important metrics in four _distinct_ colors. Knowing the number of favorites and archived devices at a glance is not essential to monitoring a network.

**Top bar**: As a new user I initially struggled to understand the purpose of items in the top bar.
<img width="574" alt="Screenshot 2024-12-03 at 09 54 22" src="https://github.com/user-attachments/assets/ce7baa15-19cb-4086-ad2e-45d3ccac92a5">
What were the buttons for? (<- -> refresh etc, full screen) Did refresh mean a new scan or just a browser (page) refresh? If the latter why are these needed when these are browser functions? As well, I found the "Process: Wait" message unintuitive (at least for me). I presume that this describes a state between scans? If so, the term "wait" is not a word to describe a state. It is an imperative (verb) or a noun (a thing). Is this telling the user to wait? The term "idle" or "paused" better describes a state or condition. In computing "idle" is well understood.

Finally, I think that the hostname and date/time could better be placed on the left like pi-hole does.

Anyway, I'm not a coder so I regret that I can only provide suggestions. I am hoping that some of these suggestions can be considered down the road for a piece of otherwise excellent open source software.

**Comment by jokob-sk** [comment]
Hi @jstars-gmti,

Thanks for the in-depth and thoughtful message!

You can choose which tiles to show in Settings:

![image](https://github.com/user-attachments/assets/e09291c6-cca6-4f9e-908a-3e2dc2ca4443)

I also changed the `Process: Wait` to `Process: Idle` in the next release. I couldn't come up with a better description of the state, so thanks for the tip! :) 

The refresh button is needed if cache needs to be refreshed - ideally that should happen automatically, but well, can't figure out why sometimes the cache is not fully initialized / cleared - I know for now this is a bit of a workaround. If somoene better than me in cache expiration reads this, I'm all ears and open to PRs :) 

I'll think about the top bar suggestions and may change things in future. 

Thanks!
j

**Comment by jstars-gmti** [comment]
Many thanks for pointing out the setting for the tiles. Really nice that everything has been designed to be configurable. 

However I have noticed that when saving any change to Settings, the UI sits at "Loading" for _several minutes_ and the status window displays "Check Plugin x of y".

<img width="413" alt="Screenshot 2024-12-03 at 13 45 25" src="https://github.com/user-attachments/assets/598aaefc-2f17-4811-bf66-8bad42538d86">

Is this a known issue or do I have something configured wrong (I only have 10 plugins loaded).

**Reply by jokob-sk** [reply]
This shows the total number of available plugins that are checked and only the one that are loaded are initialized next. I want to improve this in future to speed up loading times, but I'm nearly burned out by some refactoring work I did recently, so I'm taking things slowly.

**Reply by jokob-sk** [reply]
Also the speed of the reload depends on the system. I recommend reading this if you want to improve the performance: https://github.com/jokob-sk/NetAlertX/blob/main/docs/PERFORMANCE.md

**Reply by jstars-gmti** [reply]
Completely understood. You're doing yeoman's work and this is definitely not worth burning out on. Software refactoring is indeed unsung work, but I can imagine quite rewarding when you complete it. There will always be more work to do but it doesn't need to be done now or by you. Your focus on core functionality it is the right focus.

**Comment by jstars-gmti** [comment]
Yay! `Process: Wait` has been changed to
<img width="356" alt="Screenshot 2024-12-23 at 16 16 50" src="https://github.com/user-attachments/assets/95cc3a75-b4d0-400a-9a1b-2b3095e9aa20" />

**Reply by jokob-sk** [reply]
😄

**Reply by Coaster79** [reply]
Hello
I have the same issue with my installation 
![IDLE](https://github.com/user-attachments/assets/5d7ead69-8e4b-4e42-be9d-2557ae48d837)

I have also those error log 
![BUFFERING](https://github.com/user-attachments/assets/e2ddb854-daae-46c2-87a7-f463d92fa68a)

**Reply by jokob-sk** [reply]
Hi @Coaster79,

What's the issue? The process: Idle is expected and was requested by users. So this is by design.

If you have an issue, please submit a bug or feature request.

Thanks,
J

**Reply by Coaster79** [reply]
Hello jokob-sk,

I have several devices connected to my network, but apart from my box and the host on which I installed NetAlertX, no other device comes up. (Docker installation on Syno)

**Reply by jokob-sk** [reply]
Hi @Coaster79 ,

In 99% the issue is not setting the correct interface. See these docs how to fix this: https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md

**Reply by Coaster79** [reply]
Hello jokob-sk,

I just reviewed my config, everything works fine, thanks for the support and the response time !
